### PR TITLE
java library instrumentation doc

### LINKED ISSUE #2471: Add Libraries doc or section for Java, showing how to set up library instrumentation without an agent
There's scenarios where you may not want to/cannot use the java agent for autoinstrumentation, but _do_ want to get library instrumentation. This should be documented with code samples, similar to the .NET page: https://opentelemetry.io/docs/instrumentation/net/libraries/

### PR DESCRIPTION
**Pull request by nerudadhich**
Fixes #2471

---

**Preview**: https://deploy-preview-3568--opentelemetry.netlify.app/docs/instrumentation/java/libraries/

### CODE REVIEW

File: content/en/docs/instrumentation/java/libraries.md
Comment (laurit) [inline content/en/docs/instrumentation/java/libraries.md:173]: these are not used

Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:173]: Let me check and remove if not required.

Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md]: I have fixed this.

**Comment by cijothomas** [comment]
@nerudadhich Do you need review from .NET approvers for this or it was accidentally requested..?

**Comment by cartermp** [comment]
@cijothomas I took the .NET (and semconv) owners off

File: content-modules/semantic-conventions
Comment (svrnm) [inline content-modules/semantic-conventions:1]: We need to rebase this PR before merging to remove this. @nerudadhich No action needed, this is just a note to maintainers, we tackle this last minute, for future PRs: make sure that you always work from a synced based that has no changes to content-modules ( you can verify with `git status` or `git diff` before creating commits

File: content/en/docs/instrumentation/java/libraries.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/libraries.md:67]: Add `autoconfigure` to the cspell ignore list in the front matter (see other pages how this looks like, e.g. https://github.com/open-telemetry/opentelemetry.io/blob/main/content/en/docs/instrumentation/java/exporters.md?plain=1#L4)

File: content/en/docs/instrumentation/java/libraries.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/libraries.md:124]: add `httpclient` to the cspell ignore list as well

File: content/en/docs/instrumentation/java/libraries.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/libraries.md:150]: add `getenv` to cspell ignore list

File: content/en/docs/instrumentation/java/libraries.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/libraries.md:71]: Add a not here that `maven` can be used of course as well and that the relevant dependency here is `io.opentelemetry.instrumentation:opentelemetry-java-http-client:1.31.0-alpha`

**Review by svrnm — CHANGES_REQUESTED** [review]
Thank you! Would like to get some feedback from @open-telemetry/java-approvers as well. Here are a few initial comments by me

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:167]: done

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:68]: done

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:72]: Added, Please check.

**Review by svrnm — APPROVED** [review]
Overall lgtm, @open-telemetry/java-approvers PTAL!

File: content/en/docs/instrumentation/java/libraries.md
Comment (trask) [inline content/en/docs/instrumentation/java/libraries.md:130]: I'd recommend using [auto-configuration](https://opentelemetry.io/docs/instrumentation/java/manual/#automatic-configuration)

File: content/en/docs/instrumentation/java/libraries.md
Comment (trask) [inline content/en/docs/instrumentation/java/libraries.md:82]: (depending on outcome of https://github.com/open-telemetry/opentelemetry.io/pull/3580/files#r1406625737)
```suggestion
    <version>{{% param vers.instrumentation %}}-alpha</version>
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (trask) [inline content/en/docs/instrumentation/java/libraries.md:66]:
```suggestion
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (trask) [inline content/en/docs/instrumentation/java/libraries.md:68]:
```suggestion
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (trask) [inline content/en/docs/instrumentation/java/libraries.md:70]:
```suggestion
```

**Review by chalin — CHANGES_REQUESTED** [review]
As mentioned in https://github.com/open-telemetry/opentelemetry.io/pull/3568#discussion_r1400348752, this PR needs to be rebased. Since @svrnm has approved, I'm issuing this "request changes" as a safeguard to ensure that the PR doesn't get merged by mistake containing a retrograde of the semconv submodule.

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:28]: I'd suggest dropping the custom header anchors here and elsewhere in the added text. They aren't required, and it'll keep the markdown clean.

Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:28]: Actually, we can also probably use a more concise title for this top-level heading:

```suggestion
## Example app
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:88]:
```suggestion
### Setup
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:37]:
```suggestion
### Dependencies
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:143]:
```suggestion
### Run
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [reply content/en/docs/instrumentation/java/libraries.md:71]: @svrnm - shouldn't we use tabbed panes in this case instead?

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:161]:
```suggestion
When you run the app, the instrumentation libraries will to the following:
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:167]:
```suggestion
- If you use an instrumented HTTP server (as in the [dice app](/docs/instrumentation/java/getting-started/#example-application)),
  then more trace spans will be generated having the same trace ID
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:173]:
```suggestion
For a full list of instrumentation libraries, see [opentelemetry-java-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks).
```

**Comment by nerudadhich** [comment]
> As mentioned in [#3568 (comment)](https://github.com/open-telemetry/opentelemetry.io/pull/3568#discussion_r1400348752), this PR needs to be rebased. Since @svrnm has approved, I'm issuing this "request changes" as a safeguard to ensure that the PR doesn't get merged by mistake containing a retrograde of the semconv submodule.

Thanks @chalin , I am assuming that you will take care of rebase before merging this PR ?

**Comment by chalin** [comment]
Just to be clear, this PR needs not only a rebase, but the change to the semconv submodule will need to be manually removed. @svrnm will probably be handling all this 😄

File: content/en/docs/instrumentation/java/libraries.md
Comment (trask) [reply content/en/docs/instrumentation/java/libraries.md:71]: that would be nice here too: https://opentelemetry.io/docs/instrumentation/java/getting-started/#dependencies

Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:73]: @chalin @trask I have improved with tabbed panes now, Please suggest if any more text need to add for same ?

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [inline content/en/docs/instrumentation/java/libraries.md:41]: Adjust the mention of the filename here.

File: content/en/docs/instrumentation/java/libraries.md
Comment (theletterf) [inline content/en/docs/instrumentation/java/libraries.md:167]:
```suggestion
- Start a new trace.
- Generate a span that represents the request made to the external API endpoint.
- If you use an instrumented HTTP server, as in the
  [dice app](/docs/instrumentation/java/getting-started/#example-application), 
  more trace spans are generated with the same trace ID.
```

File: content/en/docs/instrumentation/java/libraries.md
Comment (theletterf) [inline content/en/docs/instrumentation/java/libraries.md:186]:
```suggestion
After you've set up instrumentation libraries, you might want to add
[manual instrumentation](/docs/instrumentation/java/manual) to collect custom
telemetry data.

You might also want to configure an appropriate exporter to
[export your telemetry data](/docs/instrumentation/java/exporters) to one or
more telemetry backends.

You can also check the
[automatic instrumentation for Java](/docs/instrumentation/java/automatic) for
existing library instrumentations.

```

**Review by theletterf — CHANGES_REQUESTED** [review]

**Comment by svrnm** [comment]
> goddamn submodules again

fixed that for now 🪄

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:41]: Sure @chalin . Let me try something and get back.

**Comment by theletterf** [comment]
@nerudadhich Feel free to apply the suggestions I made if you agree.

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md]: Hi @chalin I have added a note for Maven. Please suggest a change if needed.

**Comment by svrnm** [comment]
@open-telemetry/java-approvers please take a look!

**Comment by svrnm** [comment]
@open-telemetry/java-approvers please take a look!

**Comment by nerudadhich** [comment]
@open-telemetry/java-approvers Could you please review ?
cc - @trask

**Comment by trask** [comment]
hi @nerudadhich, sorry for the delay.

@jeanbisutti can you take a look at this when you have a chance? thx!

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [reply content/en/docs/instrumentation/java/libraries.md:77]: Tabbed pane seems file. I'll let @svrnm give the final work as to whether this comment is resolved to his liking.

> that would be nice here too: https://opentelemetry.io/docs/instrumentation/java/getting-started/#dependencies

@trask - maybe we can open a separate issue for this?

**Review by chalin — COMMENTED** [review]
We're getting there! See inline for a tweak to the `alert`.

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [reply content/en/docs/instrumentation/java/libraries.md]: I'm ok with the (adjusted) note (see my review comment of Jan 10).
@svrnm @theletterf - WDYT?

File: content/en/docs/instrumentation/java/libraries.md
Comment (svrnm) [reply content/en/docs/instrumentation/java/libraries.md:77]: @nerudadhich would it be a lot of effort to make this work with gradle and maven out of the box? If yes, let's keep it as is and have a separate issue for that, if no, feel free to update the PR accordingly. This would than nicely fit with the other existing documentation where we provide both

File: content/en/docs/instrumentation/java/libraries.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/libraries.md:45]: `pox.xml` should be `pom.xml`

File: content/en/docs/instrumentation/java/libraries.md
Comment (jeanbisutti) [inline content/en/docs/instrumentation/java/libraries.md:96]: Perhaps mentioned that with Spring Boot the SDK (OpenTelemetry object) can be initialized with the OpenTelemetry starter: https://opentelemetry.io/docs/languages/java/automatic/spring-boot

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:77]: Hi @svrnm I tired spending some time on this but I am not Maven expert if we can get someone's help would be great or I can still spend more time if we are not in rush.

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md:96]: Sure @jeanbisutti

**Comment by jeanbisutti** [comment]
It looks globally good to me. Perhaps this PR could be merged with the last suggestions from @chalin, and the Java library page could be improved with other PRs? @chalin @svrnm @trask What do you think?

File: content/en/docs/instrumentation/java/libraries.md
Comment (theletterf) [reply content/en/docs/instrumentation/java/libraries.md:45]: OK with it!

**Review by theletterf — APPROVED** [review]

File: content/en/docs/instrumentation/java/libraries.md
Comment (chalin) [reply content/en/docs/instrumentation/java/libraries.md:45]: LOL, fixed!

**Review by chalin — APPROVED** [review]
Once the tweak to the alert is merged, since others are ok (generally speaking) with this update, I'm giving a 👍🏻 to a merge.

I agree with @jeanbisutti that I'd prefer that we address other changes separately.

**Comment by svrnm** [comment]
@nerudadhich you need to resolve the merge conflict locally. The issue comes from the rename of `instrumentation` to `languages` via #3761. Let me know if you need help with that.

@open-telemetry/java-approvers please take another look, I think this came a long way and is ready to be merged:-)

File: content/en/docs/instrumentation/java/libraries.md
Comment (nerudadhich) [reply content/en/docs/instrumentation/java/libraries.md]: fixed.

**Comment by nerudadhich** [comment]
@svrnm I have addressed the review comments, I think we are ready to merge. Just need help to `Links` build that is failing.

**Comment by jeanbisutti** [comment]
> Once the tweak to the alert is merged, since others are ok (generally speaking) with this update, I'm giving a 👍🏻 to a merge.
> 
> I agree with @jeanbisutti that I'd prefer that we address other changes separately.

@trask Is it ok for you to merge?

**Comment by nerudadhich** [comment]
@trask Could please confirm if we can merge ? Also @svrnm How can I see report of broken links in local or somewhere so that I can fix it ?

**Comment by svrnm** [comment]
Also @svrnm How can I see report of broken links in local or somewhere so that I can fix it ?

You can run `npm run test-and-fix` locally to get most problems resolved, or it will dump you the error. You can also do `check:links` for pure link checking

**Comment by svrnm** [comment]
@open-telemetry/java-approvers please take a look

**Review by jeanbisutti — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Comment by svrnm** [comment]
Thank you for working through this, @nerudadhich, great contribution!

**Comment by nerudadhich** [comment]
> Thank you for working through this, @nerudadhich, great contribution!

Thanks for helping everyone ❤️

**Pull request merged**
